docs/versioned_docs/version-v1.3.0/02-Features/02-message-handling/03-event-hubs.md

```diff
@@ -0,0 +1,319 @@
+                .WithEventHubsMessageHandler<OrdersMessageHandler, Order>();
+
+        // Note, that only a single call to the `.WithEventHubsMessageHandler` has to be made when the handler should be used across message pumps.
+    }
+}
+```
+
+The Azure EventHubs uses the `EventProcessorClient` internally. To learn more about this way of consuming messages from Azure EventHubs, see [Microsoft's official documentation](https://docs.microsoft.com/en-us/dotnet/api/overview/azure/messaging.eventhubs.processor-readme).
+
+In this example, we are using the Azure EventHubs message pump to process event messages and use the connection strings stored in the Arcus secret store:
+- Azure EventHubs name: The name of the Event Hub that the processor is connected to, specific to the EventHubs namespace that contains it.
+- Azure EventHubs connection string secret name: The name of the secret to retrieve the Azure EventHubs connection string using your registered Arcus secret store implementation.
+- Azure EventHubs Blob storage container name: The name of the Azure Blob storage container in the storage account to reference where the event checkpoints will be stored. The events will be streamed to this storage so that the client only has to worry about event processing, not event capturing.
+- Azure EventHubs Blob storage account connection string secret name: The name of the secret to retrieve the Azure EventHubs connection string using your registered Arcus secret store implementation.
+
+### Filter messages based on message context
+When registering a new message handler, one can opt-in to add a filter on the message context which filters out messages that are not needed to be processed.
+
+This can be useful when you are sending different message types on the same queue. Another use-case is being able to handle different versions of the same message type which have different contracts because you are migrating your application.
+
+Following example shows how a message handler should only process a certain message when a property's in the context is present.
+
+We'll use a simple message handler implementation:
+
+```csharp
+using Arcus.Messaging.Abstractions;
+using Arcus.Messaging.Pumps.Abstractions.MessagingHandling;
+
+public class OrderMessageHandler : IAzureEventHubsMessageHandler<Order>
+{
+    public async Task ProcessMessageAsync(Order order, AzureEventHubsMessageContext context, ...)
+    {
+        // Do some processing...
+    }
+}
+```
+
+We would like that this handler only processed the message when the context contains `MessageType` equals `Order`.
+
+```csharp
+using Microsoft.Extensions.DependencyInjection;
+
+public class Program
+{
+    public void ConfigureServices(IServiceCollection services)
+    {
+        services.WithEventHubsMessageHandler<OrderMessageHandler, Order>(context => context.Properties["MessageType"].ToString() == "Order");
```

Comment:
If you change your example to use something like `SensorReading`or `DeviceTelemetry` you could modify the filter to filter on something like `SensorType` or `DeviceType` or something like that.  (For instance `SensorType == "speedometer"` or something)

Reply:
Ah, yeah, updated this to 'sensor status', which is probably also ok, right?

Reply:
Sure.